src/components/EventForm/EventForm.tsx

```diff
@@ -0,0 +1,279 @@
+import Button from "../Button/Button";
+import InputBox from "../InputBox/InputBox";
+import { addDoc, collection, Timestamp } from "firebase/firestore";
+import { db } from "../../firebase";
+import { useState } from "react";
+import "./EventForm.scss";
+import { useNavigate } from "react-router-dom";
+import { isBefore } from "date-fns";
+
+type DefaultForm = {
+  eventName: string;
+  eventCategory: string;
+  eventCapacity: number | null;
+  eventDescription: string;
+  eventImages: string[];
+  eventDate: Timestamp | null;
+};
+
+type EventFormProps = {
+  handleNewEvent: () => void;
+};
+
+const EventForm = ({ handleNewEvent }: EventFormProps) => {
+  const [databaseInput, setDatabaseInput] = useState<DefaultForm>({
+    eventName: "",
+    eventCategory: "",
+    eventCapacity: null,
+    eventDescription: "",
+    eventDate: null,
+    eventImages: [""],
+  });
+  const [images, setImages] = useState<string[]>([]);
+  const [inputValue, setInputValue] = useState<string>("");
+  const [validationError, setValidationError] = useState({
+    eventName: true,
+    eventCategory: true,
+    eventDate: true,
+    eventCapacity: true,
+    eventDescription: true,
+    eventImages: false,
+  });
+  const [submitted, setSubmitted] = useState<boolean>(false);
+
+  const handleEventName = (event: ChangeEvent<HTMLInputElement>) => {
+    const eventName = event.currentTarget.value;
+    setDatabaseInput({ ...databaseInput, eventName: eventName });
+    setValidationError({
+      ...validationError,
+      eventName: databaseInput.eventName.length < 1,
+    });
+  };
+  const handleEventCategory = (event: ChangeEvent<HTMLInputElement>) => {
+    const eventCategory = event.currentTarget.value;
+    setDatabaseInput({ ...databaseInput, eventCategory: eventCategory });
+    setValidationError({
+      ...validationError,
+      eventCategory: databaseInput.eventCategory.length < 1,
+    });
+  };
+  const handleEventDate = (event: ChangeEvent<HTMLInputElement>) => {
+    const inputDate = new Date(event.currentTarget.value);
+    const currentDate = new Date();
+    const isValidDate = isBefore(inputDate, currentDate);
+    const timestampDate = Timestamp.fromDate(inputDate);
+    setDatabaseInput({
+      ...databaseInput,
+      eventDate: timestampDate,
+    });
+    setValidationError({
+      ...validationError,
+      eventDate: isValidDate,
+    });
+  };
+  const handleEventCapacity = (event: ChangeEvent<HTMLInputElement>) => {
+    const eventCapacity = event.currentTarget.value;
+    if (
+      !eventCapacity.toLowerCase().search(/[a-z]/g || eventCapacity.length < 1)
+    ) {
+      setDatabaseInput({
+        ...databaseInput,
+        eventCapacity: null,
+      });
+      return;
+    }
+    setDatabaseInput({
+      ...databaseInput,
+      eventCapacity: parseInt(eventCapacity),
+    });
+
+    setValidationError({
+      ...validationError,
+      eventCapacity: databaseInput.eventCapacity === null,
```

Comment:
I think you should be checking the input from the user rather then the state, like the date there is a bit of lag between validation.